src/main/java/com/timgroup/statsd/StatsDBlockingProcessor.java

```diff
@@ -0,0 +1,102 @@
+package com.timgroup.statsd;
+
+import java.nio.ByteBuffer;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+public class StatsDBlockingProcessor extends StatsDProcessor {
+
+    private final BlockingQueue<String> messages;
+
+    StatsDBlockingProcessor(final int queueSize, final StatsDClientErrorHandler handler,
+            final int maxPacketSizeBytes, final int poolSize, final int workers)
+            throws Exception {
+
+        super(queueSize, handler, maxPacketSizeBytes, poolSize, workers);
+        this.messages = new ArrayBlockingQueue<String>(queueSize);
+    }
+
+    @Override
+    boolean send(final String message) {
+        try {
+            if (!shutdown) {
+                messages.put(message);
+                return true;
+            }
+        } catch (InterruptedException e) {
+            // NOTHING
+        }
+
+        return false;
+    }
+
+    @Override
+    public void run() {
+
+        for (int i = 0 ; i < workers ; i++) {
+            executor.submit(new Runnable() {
+                public void run() {
+                    boolean empty;
+                    ByteBuffer sendBuffer;
+
+                    try {
+                        sendBuffer = bufferPool.borrow();
+                    } catch (final InterruptedException e) {
+                        handler.handle(e);
+                        return;
+                    }
+
+                    while (!(messages.isEmpty() && shutdown)) {
+
+                        try {
+
+                            if (Thread.interrupted()) {
+                                return;
+                            }
+
+                            final String message = messages.poll(WAIT_SLEEP_MS, TimeUnit.MILLISECONDS);
```

Comment:
As several threads can poll messages,  if an application call `send("foo")`  then `send("bar")`,  the buffer containing `bar` can be queued before the one containing `foo`. 
There is no guaranty about the order of the messages. 
Is it a problem? I am thinking about the case of receiving an Event `stop` before an Event `start`.

Reply:
Indeed what you say is accurate, I had a technical discussion about this with @olivielpeau, and we don't think it's a problem, at least not in any obvious manner. I also feel that even though it's definitely true, in practice it won't matter much because the order issue would typically affect metrics submitted in a very limited timeframe. In any case, this was one of the things we had to keep track of specifically during QA.